operator/pkg/manifest/client.go

```diff
@@ -239,10 +267,12 @@ func (h *HelmClient) assignRestMapping(gvk schema.GroupVersionKind, info *resour
 }
 
 func (h *HelmClient) convertToInfo(unstructuredObj *unstructured.Unstructured) (*resource.Info, error) {
+	// manual invalidation of mem cache client to maintain the current state
+	h.mapper.Reset()
```

Comment:
Hmm not sure here. I think this should not be necessary as the restmapping assignment in line 280 will already reset the cache on failure. you can simply ignore the error and retry once if it fails instead of resetting every time. This currently will always reset the entire client cache on every install or consistency check.

Reply:
This case is only used when manifest transformers (e.g. declarative lib) decide the resources. In a standard case this is not called. I tested it, we should follow like helm does it by reseting the server mapping everytime. The other reset is only in case of an error and there is no way to discover this. I will keep it as it was since it doesn't affect what we want to achieve with this PR.

Reply:
Why should we follow like helm does it here to reset the server mapping on every install or consistency check? For me this does not make sense (even if its just in declarative library), I think we can also rely on a requeue here. Agree though that its not something that is supposed to be necessary for module manager directly

Reply:
So helm does it AFAIK to make sure helm repo update resource consistencies for rest mapping in the discovery client. As long as we offer static charts through OCI layers it is ok, but the remote helm repo (backward compatibility) case is what which will be affected by this reset. As this is off-topic let's discuss it outside this issue.